### PR TITLE
Social options

### PR DESCRIPTION
**Pull request by zacheryconverse**
# Submit a pull request

### 🎯 Goal

Style message options, replies, and reaction selector

### 🛠 Implementation details

New ThreadReply directory and scss

### 🎨 UI Changes

Screen Shot 2021-10-01 at 1 08 17 PM

### CODE REVIEW

File: examples/website-demos/social-messaging/src/assets/LeftReply.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const LeftReply: React.FC = () => (
+  <svg width="17" height="33" viewBox="0 0 17 33" fill="none" xmlns="http://www.w3.org/2000/svg" style={{ position: 'relative', bottom: '6px' }}>
+    <path d="M0.5 32V32C9.33656 32 16.5 24.8366 16.5 16L16.5 0" stroke="#DBDDE1"/>
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/LeftReply.tsx:3]: do the colors on these reply icons need to be variables?

File: examples/website-demos/social-messaging/src/components/Message/SocialMessageUI.tsx [hunk 2]
```diff
@@ -99,27 +148,30 @@ export const SocialMessage: React.FC<
...
+              readByMembers?.length < 1 && <DeliveredCheckmark />}
+            {myMessage && readByMembers && readByMembersLength && <DoubleCheckmark />}
+            {!myMessage && (
+              <div className='message-wrapper-inner-data-info'>
+                {message.user?.name || message.user?.id}
+              </div>
+            )}
+            <MessageTimestamp customClass='message-wrapper-inner-data-time' />
+          </div>
+          {!myMessage && <MessageOptions />}
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/Message/SocialMessageUI.tsx:174]: do we need two references to MessageOptions? i'm assuming this is absolutely positioned so can't we do...
`<MessageOptions displayReplies={myMessage} />`

File: examples/website-demos/social-messaging/src/components/ThreadReply/ThreadReply.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const uniqueThreadParticipants = Array.from(new Set(message.thread_participants)).slice(0, 2);
+
+  return (
+    <>
+      {message.thread_participants ? (
+      <div className='message-wrapper-inner-reply'>
+        {!myMessage && (
+          <>
+            <RightReply />
+            {uniqueThreadParticipants[0] && uniqueThreadParticipants.map((user, i) => (
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ThreadReply/ThreadReply.tsx:24]: what happens if a thread has a lot of participants? we should limit to 4 or so

File: examples/website-demos/social-messaging/src/assets/LeftReply.tsx (hunk 1 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/assets/LeftReply.tsx:3]: Oh yeah good catch

File: examples/website-demos/social-messaging/src/components/ThreadReply/ThreadReply.tsx (hunk 3 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/components/ThreadReply/ThreadReply.tsx:24]: line 15 slices the number of participants to 2. I didnt see anything in the design with more than 2. I can increase to 4 if you like that better?

File: examples/website-demos/social-messaging/src/components/Message/SocialMessageUI.tsx (hunk 2 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/components/Message/SocialMessageUI.tsx:174]: there was no need for 2 - I removed 1 and adjusted with css

**Review by DanC5 — APPROVED** [review]

**Pull request merged**